src/dispatch/plugins/dispatch_slack/modals.py

```diff
@@ -482,3 +498,132 @@ def update_notifications_group_from_submitted_form(action: dict, db_session=None
     group_plugin = plugins.get(INCIDENT_PLUGIN_GROUP_SLUG)
     group_plugin.add(incident.notifications_group.email, members_added)
     group_plugin.remove(incident.notifications_group.email, members_removed)
+
+
+def build_add_timeline_event_blocks(incident: Incident):
+    """Builds all blocks required to add an event to the incident timeline."""
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Add Timeline Event"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "plain_text",
+                        "text": "Use this form to add an event to the incident timeline.",
+                    }
+                ],
+            },
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Add"},
+        "callback_id": AddTimelineEventCallbacks.submit_form,
+        "private_metadata": json.dumps({"incident_id": str(incident.id)}),
+    }
+
+    date_picker_block = {
+        "type": "input",
+        "block_id": AddTimelineEventBlockFields.date,
+        "label": {"type": "plain_text", "text": "Date"},
+        "element": {"type": "datepicker"},
+    }
+    modal_template["blocks"].append(date_picker_block)
+
+    hour_picker_options = []
+    for h in range(0, 24):
+        hour_picker_options.append(create_block_option_from_template(text=h, value=str(h).zfill(2)))
```

Comment:
Is zfill used to handle the case when no hour is selected? Can the field not have a default value instead?

Reply:
No, `zfill` only fills out the value with zeros until it reaches the character length provided. Let me see if I can make the field required, and if not, if I can add a default.